### ISSUE TITLE
gumbel softmax in keras

### ISSUE THREAD

**Issue by Johnxjp** [keras-team/keras#10938]
I'm trying to build a seq2seq model that generates sentences and classifies them. To do the latter, I want to feed the decoder's softmax output to a pre-trained classifier. This classifier however uses a Keras embedding layer so passing the raw softmax into the classifier isn't an option. I thought I could use a gumbel softmax to get a one-hot encoding then use the OneHotEmbedding layer I found here (https://github.com/keras-team/keras/issues/2505) to solve this.

Eric Jang provided this TensorFlow code for the gumbel softmax and I wanted to know how to turn this into a Keras layer. In particular, I am interested in the `hard` property that ensures the vector on the forward pass is strictly categorical but on the backward pass, the gradient is the gumbel softmax output. Can anyone help with this please? 

Thanks.
